### PR TITLE
Carrier portal features

### PR DESCRIPTION
**Pull request by j8seangel**
This PR includes
- New array prop called `markers` to render the desired content on top of the map at lat and long.
- Fix custom gl layers, so now we are able to render any layer following the mapbox spec
- Typescript definition types for the exported components.
However, it still doesn't work for code splitting imports
image

So you have to include the following code on the project you want to use them:
```
declare module '@globalfishingwatch/map-components/components/countryflag' {
  import { CountryFlag } from '@globalfishingwatch/map-components'
  export = CountryFlag
}
```

### CODE REVIEW

File: src/map/glmap/Map.container.js [hunk 1]
```diff
@@ -23,8 +25,23 @@ const getInteractiveLayerIds = createSelector(
   [getStaticLayers],
   // Note: here we assume that layer IDs provided with module match the GL layers that should
   // be interactive or not, ie typically the fill layer if a label layer is present
-  (staticLayers) =>
-    staticLayers.filter((l) => l.interactive === true && l.visible === true).map((l) => l.id)
+  (staticLayers) => {
+    return staticLayers.reduce((acc, layer) => {
+      if (!layer.interactive || !layer.visible) return acc
+      // We also need to check nested layers interactivity when custom gl layers are provided
```
[the remark this answers is not in the document]
Comment (nerik) [reply src/map/glmap/Map.container.js]: Speaking of which: https://github.com/GlobalFishingWatch/map-components/issues/71

File: src/map/glmap/Map.container.js [hunk 2]
```diff
@@ -23,8 +25,23 @@ const getInteractiveLayerIds = createSelector(
...
+      if (layer.gl !== undefined) {
+        layer.gl.layers.forEach((glLayer, index) => {
+          if (index === 0 || glLayer.interactive === true) {
```
Comment (nerik) [inline src/map/glmap/Map.container.js]: Really not sure about `if (index === 0 `
It should be explicit, not magic.

File: src/map/glmap/Map.js [hunk 3]
```diff
@@ -208,6 +217,7 @@ Map.propTypes = {
   transitionEnd: PropTypes.func,
   cursor: PropTypes.string,
   hasHeatmapLayers: PropTypes.bool.isRequired,
+  markers: PropTypes.array,
```
Comment (nerik) [inline src/map/glmap/Map.js]: Would you consider adding detail here using `arrayOf` ? Or as a rule do we consider ts definitions sufficient?

File: src/map/index.js [hunk 4]
```diff
@@ -1,3 +1,9 @@
 export { default } from './map'
 export { targetMapVessel } from './store'
 export { default as AVAILABLE_BASEMAPS } from './basemaps'
+// export {
```
Comment (nerik) [inline src/map/index.js]: Remove?

File: src/map/glmap/Map.js (hunk 3 above)
Comment (j8seangel) [reply src/map/glmap/Map.js]: Good catch, added here: https://github.com/GlobalFishingWatch/map-components/commit/6d64577650294bfd172ed42e386d12a3b11ada82

Comment (j8seangel) [reply src/map/glmap/Map.js]: added the fix and we can discuss how to manage ts and proptypes definitions later

File: src/map/glmap/Map.container.js [hunk 5]
```diff
@@ -31,7 +29,9 @@ const getInteractiveLayerIds = createSelector(
       // We also need to check nested layers interactivity when custom gl layers are provided
       if (layer.gl !== undefined) {
         layer.gl.layers.forEach((glLayer, index) => {
-          if (index === 0 || glLayer.interactive === true) {
+          // layers.length === 1 is used to ensure when parent layer is marked as interactive we have to
+          // have at least one interactive sublayer, then checked eah one individually
+          if (layer.gl.layers.length === 1 || glLayer.interactive === true) {
```
Comment (nerik) [inline src/map/glmap/Map.container.js]:
```
if (layer.gl.layers.length === 1 || layer.interactive === true) {
```

File: src/map/glmap/Map.container.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (j8seangel) [reply src/map/glmap/Map.container.js]: 👍

**Pull request merged**